### PR TITLE
Hooks app tools part

### PR DESCRIPTION
**Pull request by DonnaWuDongxia**
Hi, Robert. I have wend through the process to get a bundle.
My test sample:
https://drive.google.com/a/intel.com/file/d/0B_e_Mp4JYfsreHV2SjRDd2RzaEU/view?usp=sharing

### CODE REVIEW

**Comment by r0b5t4** [comment]
@DonnaWuDongxia  I don't understand why you need to have several extensions in a single directory. Could they not be placed in separate dirs with hooks of their own?

**Comment by DonnaWuDongxia** [comment]
@r0b5t4 I did this because multiple extensions for Crosswalk-windows can be holed in one folder currently. A Windows extension binary is just one DLL, no folder for it.

**Comment by r0b5t4** [comment]
@DonnaWuDongxia but it seems weird to me breaking a nice scheme one folder per extensions, and then work around it with multiple hooks files.

Can you implement everything in a single hooks file maybe? Why do you need more than one? Maybe you could create a loop in prePackage() / postPackage()?

**Comment by DonnaWuDongxia** [comment]
@r0b5t4 Of cause, we can require the app developer to separate the extensions and each one takes a folder. But there are some differences between Android extensions and other platfroms. Each Android extension should be a folder and extension for others platform are single lib file, please refer: https://github.com/crosswalk-project/crosswalk-website/wiki/Writing-a-Crosswalk-Extension#loading-the-extensions 
On platforms besides Android, if we require each extension take one folder, it means following points:
1. if no hooks for the extensions(normal cases), then each folder will just contains one lib binary, DLL on Windows.
2. the name of the extension folder will be meaningless, while the folder name of Android extension is actually the extension name.
3. all the separated folders will still be merged into one folder, because, crosswalk just take one directory and get direct lib files as extensions.

**Comment by r0b5t4** [comment]
@DonnaWuDongxia so what about implementing a loop in the prePackage() / postPackage() methods? I would prefer to support that, with changes if needed. The approach with multiple hooks files does not look very elegant to me. Many thanks.

**Comment by DonnaWuDongxia** [comment]
@r0b5t4 so we should keep just one root hooks in each extension folder, right?

**Comment by r0b5t4** [comment]
@DonnaWuDongxia yes, I would really prefer that, if possible. We can add extra parameters if needed, but I am not very happy with multiple hooks files.

**Comment by DonnaWuDongxia** [comment]
@r0b5t4 OK, I will modify it tomorrow. Thanks!

**Comment by DonnaWuDongxia** [comment]
@r0b5t4 I have seperated the extensions and their hooks into folders as we discussed tomorrow. With the updates, hooks will be skipped from the msi package. My test sample for this enable code:
https://drive.google.com/a/intel.com/file/d/0B_e_Mp4JYfsrWU4wVzlybFpIVlU/view?usp=sharing

**Comment by r0b5t4** [comment]
@DonnaWuDongxia not sure i understand. My suggestion was to use a single XWalkExtensionHooks.js, and loop all extensions in its prePackage() / postPackage() methods. I'm not seeing a hooks file in the test sample above.

**Comment by DonnaWuDongxia** [comment]
There are XWalkExtensionHooks.js file in each extension folder, realsense/ep/, etc.

**Comment by r0b5t4** [comment]
But my suggestion was to use a single XWalkExtensionHooks.js in the toplevel directory. Otherwise we will have to special-case android/windows in app-tools, which is not nice. Why can you not implement all the funtionality in a single XWalkExtensionHooks.js?

**Comment by DonnaWuDongxia** [comment]
@r0b5t4 we did keep only one hook file in each extension folder.

**Comment by r0b5t4** [comment]
oh, so the toplevel folder from the sample is not the folder you specify in the manifest?

**Comment by DonnaWuDongxia** [comment]
@r0b5t4 thanks!

File: src/util/index.js [hunk 1]
```diff
@@ -25,6 +25,15 @@ var util = {
     /** {@link Version} */
     Version: require("./Version"),
 
+    /** expose some dependencies for hooks */
```
Comment (r0b5t4) [inline src/util/index.js]: Those are standard node modules. You should not need to export them with our internal mechanism. Otherwise everything LGTM!

Comment (DonnaWuDongxia) [inline src/util/index.js]: Yes, they are node modules. But not build-in modules, need to be installed. Hooks are executed from temporary dir in which these modules can't be required.

**Comment by r0b5t4** [comment]
Ok, not ideal, but ok.

**Comment by DonnaWuDongxia** [comment]
Thanks!

**Comment by DonnaWuDongxia** [comment]
@r0b5t4  Then, when could this branch be merged  to master?

**Comment by r0b5t4** [comment]
@DonnaWuDongxia done.

**Pull request merged**